containerd-http-proxy/configure_http_proxy.yaml

```
@@ -0,0 +1,88 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: containerd-http-proxy
+spec:
+  selector:
+    matchLabels:
+      name: containerd-http-proxy
+  template:
+    metadata:
+      labels:
+        name: containerd-http-proxy
+    spec:
+      hostPID: true
+      containers:
+      - name: startup-script
+        image: gke.gcr.io/debian-base:bookworm-v1.0.0-gke.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          set -u
+
+          validate_proxy() {
+            input_string=$1
+
+            if echo "$input_string" | grep -q ' '; then
+                echo "Error: Input cannot contain spaces. Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | sed 1d | grep -q .; then
+                echo "Error: Input cannot contain newline. Input: '$input_string'"
+                exit 1
+            fi
+
+            if echo "$input_string" | grep -q -e '"' -e "'"; then
+              echo "Error: Input cannot contain quotes. Input: '$input_string'"
+              exit 1
+            fi
+          }
+
+          validate_proxy "${HTTP_PROXY:-}"
+          validate_proxy "${HTTPS_PROXY:-}"
+          validate_proxy "${NO_PROXY:-localhost}"
+
+          STARTUP_SCRIPT=$(cat <<'END'
+          set -e
+          set -u
+          mkdir -p /etc/systemd/system/containerd.service.d
+          cat > /etc/systemd/system/containerd.service.d/http-proxy.conf <<EOF
+          [Service]
+          Environment="HTTP_PROXY=${HTTP_PROXY}"
+          Environment="HTTPS_PROXY=${HTTPS_PROXY}"
+          Environment="NO_PROXY=${NO_PROXY:-localhost}"
```

Consider escaping these for added safety:
```suggestion
          $(printf 'Environment="HTTP_PROXY=%q"' "${HTTP_PROXY}")
          $(printf 'Environment="HTTPS_PROXY=%q"' "${HTTPS_PROXY}")
          $(printf 'Environment="NO_PROXY=%q"' "${NO_PROXY:-localhost}")
```

---

%q seems not supported by dash. We had to use dash because bash is not included in the image.